src/adr/0041-james-servers-offering.md

```diff
@@ -0,0 +1,109 @@
+the behaviour of Apache James.
+
+ - **Distributed Mail server**: scaling mail storage infrastructure is notoriously hard as most 
+mail server leverages only the file system. James project delivers a distributed mail server 
+leveraging modern noSQL infrastructure to ease scaling.
+
+### Current situation
+
+We currently deliver as part of the Apache James server:
+
+ - **Spring server** (`server/app`): barely maintained, Spring server exposes configuration allowing 
+class level overides, resulting in untested componant combinaison leading to unexpected runtime errors. 
+Its usage requires configuration to adapt it to the user use cases, wich can be 
+not obvious for people with limited knowledge/resources, and can discourage self hosting usages.
+
+Guice servers includes:
+
+ - **Cassandra - guice** (`server/container/guice/cassandra-guice`) and its LDAP counter-part
+(`server/container/guice/cassandra-guice-ldap`): about to be deprecated, this guice server relies on
+a Cassandra & ElasticSearch database without achieving distribution (as it misses a messaging component).
+
+ - **Distributed James** (`server/container/guice/cassandra-rabbitmq-guice`) and its LDAP counter part
+(`server/container/guice/cassandra-rabbitmq-ldap-guice`) enabled to achieve the distributed use case. This
+server furthermore is extensible, and ships many extra features by default. It is however hard to operate.
+
+ - **Guice memory** (`server/container/guice/memory-guice`) ships a server based on memory. Easy to boot, 
+without external dependencies, we mostly rely on it for testing purposes.
+
+ - **JPA Guice** (`server/container/guice/jpa-guice`) ships an extensible server based on JPA storage. With
+a default configuration using Derby embedded database, this server have no external dependencies by default.
+However the still large amount of configuration can make new comers confused. It currently satisfies operator 
+looking for self hosting solutions as well as those looking for an extensible server.
+
+ - **JPA SMTP** (`server/container/guice/jpa-smtp`) demonstrate a Mail Transfer Agent, dedicated to mail
+processing, that do not ship the James Mailbox, as well as Mail Delivery protocols like IMAP, JMAP or POP3.
+
+We can notice that this offer is specified in technical terms, and do not directly map to identified user needs.
+
+Also the large amount of servers makes the choice harder to make for newcomers.
+
+## Decision
+
+We should deliver a subset of mail server directly adressing users needs, and brand them according to their use cases
+and not due to their technical details.
+
+In order to get a clearer offer we should first reduce the server cardinality. This can be achieved via:
+
+ - **Spring server** deprecation and removal
+ - Merging Guice servers with their LDAP variations (see [ADR 36 about guice module choosing](0036-against-use-of-conditional-statements-in-guice-modules.md))
+ - **Cassandra-guice** deprecation and removal
```

Comment:
We have to provide a decent substitution first (JPA is not a good substitution for now)

Reply:
>  We have to provide a decent substitution first (JPA is not a good substitution for now)

If it is non consensual then let's forget this deprecation. 

I would like us to think of a non technical branding for **Cassandra-guice**.

Also, I'm intending to present in this document a short term (3 month, with a delivery hopefully in 3.6.0) achievable offerings. Things not following this temporality will need IMO to be discussed separately.

Reply:
> We have to provide a decent substitution first (JPA is not a good substitution for now)

@mbaechler Probably fit for an other ADR, but could you explain quickly why?

Reply:
I don't want my data in derby and my indices into Lucene files for a small org (around 100 users), even if Cassandra and ES are not perfect they are way more reliable than files.
And features are lagging (no blob-store, limited support for search, etc)